### PR TITLE
Ensure FIRLogger initialization

### PR DESCRIPTION
**Pull request by paulb777**
Internal b/111834543

### CODE REVIEW

**Comment by wilhuff** [comment]
FWIW I think this indicates a broken relationship between GULLogger and FIRLogger.

Several customers have dinged us for having +load methods because the contribute to a longer startup time.

This is also an arms race: if something that calls GULLogger is called during +load somewhere else the ordering of these isn’t defined so this doesn’t guarantee a fix.

Better to make GULLogger’s initialization method something that can be called more than once perhaps allowing FIRLogger to specify that it trumps the generic configuration.

File: Firebase/Core/FIROptions.m [hunk 1]
```diff
@@ -109,6 +110,12 @@ + (FIROptions *)defaultOptions {
 
 #pragma mark - Private class methods
 
++ (void)load {
```
Comment (baolocdo) [inline Firebase/Core/FIROptions.m]: I'm not sure if the order of the classes is always that FIRLogger is before Analytics classes as classes might get registered at different order at runtime.

**Comment by paulb777** [comment]
@wilhuff I agree with your concerns, but I don't know of another solution to keep compatible behavior that can be done reliably.  FIRLogger requires Firebase flags to be checked before initializing the asl logger.  Firebase may not be initialized before GULLogger gets called by something else. More context in b/111834543.

**Comment by paulb777** [comment]
Closing in favor of #1592

**Pull request closed without merging**